### PR TITLE
gnu-apl 1.7 (add postgresql option)

### PR DESCRIPTION
**Pull request by fusiongyro**

### CODE REVIEW

**Comment by ilovezfs** [comment]
I think using `libpq` instead will be sufficient.

File: Formula/gnu-apl.rb [hunk 1]
```diff
@@ -20,20 +20,34 @@ class GnuApl < Formula
...
 
+    args = %W[
+      --disable-debug
+      --disable-dependency-tracking
+      --disable-silent-rules
+      --prefix=#{prefix}
+    ]
+
+    if build.with? "postgresql"
+      ENV.prepend "LDFLAGS", "-L#{Formula["postgresql"].opt_lib}"
```
Comment (ilovezfs) [inline Formula/gnu-apl.rb]: instead `args << "--with-postgresql=#{Formula["libpq"].opt_bin}/pg_config"`

Comment (ilovezfs) [reply Formula/gnu-apl.rb]: and actually it defaults to on so you don't even need to pass this argument to configure or set any of the flags manually

**Comment by ilovezfs** [comment]
@fusiongyro ping

**Comment by fusiongyro** [comment]
@ilovezfs I guess the patch isn't needed if it detects Postgres if you just build from source. Isn't that what you were saying? I should close the pull request then, right?

**Comment by ilovezfs** [comment]
@fusiongyro no I meant that the optional dependency is sufficient and it will be detected without messing with the configure options. But it's not going to find it unless it's explicitly specified as an option in the formula.

**Comment by fusiongyro** [comment]
Then the best course of action would be for me to back out all the changes except adding that optional dependency on libpq, right? I apologize for needing to be spoon-fed here, this is my first MacPorts PR.

**Comment by ilovezfs** [comment]
>Then the best course of action would be for me to back out all the changes except adding that optional dependency on libpq, right?

Yep.

>MacPorts

s/MacPorts/Homebrew/ :)

**Comment by fusiongyro** [comment]
I tried this, which I think is what you were suggesting, but it doesn't seem to pick up the PostgreSQL support. I'm wondering if it works differently for you or if I missed a step.

**Comment by ilovezfs** [comment]
```
checking for pg_config... /usr/local/opt/libpq/bin/pg_config
checking for PostgreSQL libraries... yes
checking for PostgreSQL usability... yes
```
and
```
iMac-TMP:~ joe$ brew linkage gnu-apl
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libc++.1.dylib
  /usr/lib/libncurses.5.4.dylib
  /usr/lib/libsqlite3.dylib
Homebrew libraries:
  /usr/local/opt/libpq/lib/libpq.5.dylib (libpq)
Dependencies with no linkage:
  readline
```
So I'm not sure what you mean.

**Comment by ilovezfs** [comment]
Are you building with this command?
```
brew install --with-libpq gnu-apl
```

**Comment by fusiongyro** [comment]
No, and additionally, though I have Postgres installed, I apparently did not have libpq installed. Odd.

Since using the method you suggested with this one-line change, it works for me, I am able to connect to Postgres via GNU APL.

**Comment by ilovezfs** [comment]
>No, and additionally, though I have Postgres installed, I apparently did not have libpq installed. Odd.

They are deliberately disjoint so that people can use the bottles with whichever PostgreSQL they want. If we use `depends_on :postgresql` then the bottles will use the `postgresql` formula (but with a source build you could still use whichever PostgreSQL you wanted).

We could probably find a way to get the `postgresql` to itself depend on the libpq formula but that would just be a minor optimization in terms of disk space.

**Comment by ilovezfs** [comment]
@fusiongyro Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**